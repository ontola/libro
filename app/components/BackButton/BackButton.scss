@import '../shared/config';

.BackButton {
  color: color('grey', 'mid-light');
  display: block;
  font-size: font-size('small');
  padding: units('small', 'none');

  &:hover {
    color: color('grey', 'mid-dark');
  }
}
