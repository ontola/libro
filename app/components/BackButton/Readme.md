    <BackButton link="#">Ga terug naar het overzicht</BackButton>
