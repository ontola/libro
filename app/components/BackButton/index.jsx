import PropTypes from 'prop-types';
import React from 'react';
import FontAwesome from 'react-fontawesome';

import Link from '../Link';

import './BackButton.scss';

const propTypes = {
  children: PropTypes.node,
  link: PropTypes.string.isRequired,
};

const BackButton = ({
  children,
  link,
}) => (
  <Link className="BackButton" to={link}>
    <FontAwesome name="th" />{' '}
    {children}
  </Link>
);

BackButton.propTypes = propTypes;

export default BackButton;
