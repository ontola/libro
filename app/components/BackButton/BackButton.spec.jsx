import { shallow } from 'enzyme';
import React from 'react';
import { StaticRouter } from 'react-router';

import BackButton from './index';

const comp = () => shallow((
  <StaticRouter context={{}}>
    <BackButton link="http://argu.co/">Go back</BackButton>
  </StaticRouter>
));

describe('BackButton', () => {
  it('renders', () => {
    expect(comp()).toMatchSnapshot();
  });

  it('sets the link', () => {
    expect(shallow(<BackButton link="a" />)).toHaveProp('to', 'a');
  });
});
