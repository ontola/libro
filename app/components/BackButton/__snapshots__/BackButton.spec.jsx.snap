// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <BackButton
    link="http://argu.co/"
  >
    Go back
  </BackButton>
</Router>
`;
